armsrc/lfsampling.c

```diff
@@ -424,6 +425,107 @@ uint32_t SampleLF(bool verbose, uint32_t sample_size, bool ledcontrol) {
     BigBuf_Clear_ext(false);
     return ReadLF(true, verbose, sample_size, ledcontrol);
 }
+
+/**
+* SampleLF()/SniffLF() + ReadLF() + DoAcquisition_config() 
+* @return sampling result
+**/
+int ReadLF_realtime(bool reader_field) {
+    BigBuf_Clear_ext(false);
+    LFSetupFPGAForADC(config.divisor, reader_field);
+
+    // parameters from config and constants
+    const uint8_t bits_per_sample = config.bits_per_sample;
+    const int16_t trigger_threshold = config.trigger_threshold;
+    int32_t samples_to_skip = config.samples_to_skip;
+    const uint8_t decimation = config.decimation;
+
+    uint32_t sample_size = 64;
+    const int8_t size_threshold_table[9] = {0, 64, 64, 48, 64, 40, 48, 56, 64};
+    const int8_t size_threshold = size_threshold_table[decimation];
+
+    // DoAcquisition() start
+    uint8_t last_byte = 0;
+    uint8_t curr_byte = 0;
+    int return_value = PM3_SUCCESS;
+
+    initSampleBuffer(&sample_size); // sample size in bytes
+    sample_size <<= 3; // sample size in bits
+    sample_size /= bits_per_sample; // sample count
+
+    bool trigger_hit = false;
+    int16_t checked = 0;
+
+    while (BUTTON_PRESS() == false) {
+
+        // only every 4000th times, in order to save time when collecting samples.
+        // interruptible only when logging not yet triggered
+        if ((checked >= 4000) && trigger_hit == false) {
```

Comment:
Got it

Reply:
and we rarely use specialized  compiler directives,   since we support several OS/compiler versions.   

Cluttering code is _not_  not what we want.

Reply:
> and we rarely use specialized compiler directives, since we support several OS/compiler versions.
> 
> Cluttering code is _not_ not what we want.

I think the GCC extensions there is acceptable because we only use `gcc-arm-none-eabi` to compile the code. And the code won't be too messy
from `if ((checked >= 4000) && trigger_hit == false) {`
to `if (unlikely(trigger_hit == false && (checked >= 4000))) {`
@iceman1001 How do you think? The `unlikely()` does balance the length of each branch